Add error message for using call expression in case guard clause (#2511)
gleam-lang/gleam: CHANGELOG.md

### Language changes

> - Added a better error message for situation when user tries to use call expression in case `if` guard.

Reviewer: ```suggestion
- The error message presented when a function is called in a guard has been improved.
```
